app/controllers/mainController.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-restricted-globals */
+/* eslint-disable consistent-return */
+// Utilisation de PSQL
+
+const dataMapper = require('../dataMappers/filmDataMapper');
+const userDataMapper = require('../dataMappers/userDataMapper');
+const formatUser = require('../utils/formatUser');
+
+const mainController = {
+
+  renderHomePage(_, res) {
+    res.render('home');
+  },
+
+  redirectHomePage(_, res) {
+    res.redirect('/app');
+  },
+
+  async renderFilmsListPage(_, res) {
+    try {
+      const films = await dataMapper.getdAllFilms();
+      res.render('films', { films });
+    } catch (err) {
+      res.status(500).send('Server Error');
+    }
+  },
+
+  async renderFilmsByReleaseDatePage(req, res) {
+    try {
+      const param = req.params.filmDate;
+
+      if (isNaN(param) === true || param === undefined) {
```

Comment:
```suggestion
      if (isNaN(param)) {
```

ça suffit déjà ici, car ça va catcher le cas `undefined`, pas besoin de le traiter à part